cime/src/drivers/mct/main/prep_glc_mod.F90

```diff
@@ -607,6 +777,156 @@ end subroutine prep_glc_map_one_state_field_lnd2glc
+    ! Also could map o2x_ox->o2x_gx(1) but using x2g_gx as destination allows us to see
+    ! these fields on the GLC grid of the coupler history file, which helps with debugging.
+    call seq_map_map(mapper_So2g, o2x_ox, x2g_gx, &
+    fldlist='So_blt:So_bls:So_htv:So_stv:So_rhoeff',norm=.true.)
+
+    ! inputs to melt flux calculation
+    index_x2g_So_blt =    mct_avect_indexra(x2g_gx,'So_blt',perrwith='quiet')
+    index_x2g_So_bls =    mct_avect_indexra(x2g_gx,'So_bls',perrwith='quiet')
+    index_x2g_So_htv =    mct_avect_indexra(x2g_gx,'So_htv',perrwith='quiet')
+    index_x2g_So_stv =    mct_avect_indexra(x2g_gx,'So_stv',perrwith='quiet')
+    !index_x2g_So_rhoeff = mct_avect_indexra(x2g_gx,'So_rhoeff',perrwith='quiet')
+
+    index_g2x_Sg_tbot =   mct_avect_indexra(g2x_gx,'Sg_tbot',perrwith='quiet')
+    index_g2x_Sg_dztbot = mct_avect_indexra(g2x_gx,'Sg_dztbot',perrwith='quiet')
+    index_g2x_Sg_lithop = mct_avect_indexra(g2x_gx,'Sg_lithop',perrwith='quiet')
+    index_g2x_Sg_icemask_floating = mct_avect_indexra(g2x_gx,'Sg_icemask_floating',perrwith='quiet')
+
+    ! outputs to melt flux calculation
+    index_g2x_Sg_blis = mct_avect_indexra(g2x_gx,'Sg_blis',perrwith='quiet')
+    index_g2x_Sg_blit = mct_avect_indexra(g2x_gx,'Sg_blit',perrwith='quiet')
+    index_g2x_Fogx_qiceho = mct_avect_indexra(g2x_gx,'Fogx_qiceho',perrwith='quiet')
+    index_g2x_Fogx_qicelo = mct_avect_indexra(g2x_gx,'Fogx_qicelo',perrwith='quiet')
+    index_x2g_Fogx_qiceli = mct_avect_indexra(x2g_gx,'Fogx_qiceli',perrwith='quiet')
+    index_x2g_Fogx_qicehi = mct_avect_indexra(x2g_gx,'Fogx_qicehi',perrwith='quiet')
+
+    gsize = mct_aVect_lsize(g2x_gx)
+    write(logunit,*) 'gsize=',gsize
+
+    do n=1,gsize
+      !Extract glc and ocn-sourced coupler fields used as input to compute_melt_fluxes to local arrays...
+
+      ! Fields from the ocean, now on the GLC grid
+      oceanTemperature(n) =               x2g_gx%rAttr(index_x2g_So_blt,n)
+      oceanSalinity(n) =                  x2g_gx%rAttr(index_x2g_So_bls,n)
+      oceanHeatTransferVelocity(n) =      x2g_gx%rAttr(index_x2g_So_htv,n)
+      oceanSaltTransferVelocity(n) =      x2g_gx%rAttr(index_x2g_So_stv,n)
+
+      ! Fields from the ice sheet model (still on the GLC grid)
+      iceTemperature(n) =                 g2x_gx%rAttr(index_g2x_Sg_tbot,n)
+      iceTemperatureDistance(n) =         g2x_gx%rAttr(index_g2x_Sg_dztbot,n)
+      interfacePressure(n) =              g2x_gx%rAttr(index_g2x_Sg_lithop,n)
+      iceFloatingMask(n) =                g2x_gx%rAttr(index_g2x_Sg_icemask_floating,n)
+
+      !... initialize local compute_melt_fluxes output arrays...
+      outInterfaceSalinity(n)     =       0.0_r8
+      outInterfaceTemperature(n)  =       0.0_r8
+      outFreshwaterFlux(n)  =             0.0_r8
+      outOceanHeatFlux(n) =               0.0_r8
+      outIceHeatFlux(n) =                 0.0_r8
+    end do
+
+    !Fyke: hardwire in some *reasonable* values here to test things out.
```

Comment:
More commented-out code to remove.

Reply:
Fixed in 733a46925b5580e5888bdfa417add9916e7cf2df